### PR TITLE
CP2K support for dpgen2

### PR DESCRIPTION
**Pull request by Andy6M**
## Summary by CodeRabbit

- **New Features**
  - Introduced support for CP2K simulations with new classes for input preparation and execution.
  - Added configuration files for water simulations using different tools.
  
- **Bug Fixes**
  - Simplified `execute` function in the `Abacus` class by removing an unnecessary argument.

- **Tests**
  - Added test cases to validate the new CP2K functionalities.
  - Included input files for water simulations in the test suite.

- **Dependencies**
  - Added `cp2kdata` to the list of project dependencies.

### CODE REVIEW

File: dpgen2/fp/cp2k.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    @OP.exec_sign_check
+    def execute(
+        self,
+        ip: OPIO,
+    ) -> OPIO:
+        confs = []
+        # remove atom types with 0 atom from type map
+        # for all atom types in the type map
+        for p in ip["confs"]:
+            for f in p.rglob("type.raw"):
```
Comment (wanghan-iapcm) [inline dpgen2/fp/cp2k.py:74]: Optional revision: recommended to use `dpdata.MultiSystems` to handle the confs, then you do not need to manually handle the type mappings.  
```python
ms = dpdata.MultiSystems(type_map=type_map) 
ms.from_deepmd_npy(p, labeled=False)
for ii in range(len(ms)): 
    s = ms[ii]                                                                                                                                              
```

File: tests/fp/test_cp2k.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        if Path("upload").is_dir():
+            shutil.rmtree("upload")
+        if Path("output").is_dir():
+            shutil.rmtree("output")
+        for p in Path(".").glob("cp2k-dpgen-*"):
+            shutil.rmtree(p)
+
+    def test_cp2k(self):
+        data_path = Path(__file__).parent / "data.cp2k"
+        print(data_path)
```
[the remark this answers is not in the document]
Comment (Andy6M) [reply tests/fp/test_cp2k.py:38]: Thank you for your suggestion.

File: dpgen2/fp/cp2k.py (hunk 1 above)
Comment (Andy6M) [reply dpgen2/fp/cp2k.py:74]: Hi @wanghan-iapcm,

Thanks for the suggestion. I prefer to keep the current implementation for now.

Comment (wanghan-iapcm) [reply dpgen2/fp/cp2k.py:74]: that's no problem.

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]
Could you please add an example, showcasing how to use cp2k in dpgen2?

**Comment by Andy6M** [comment]
> Could you please add an example, showcasing how to use cp2k in dpgen2?

Thank you for your suggestion. Two example files, `examples/water/input_dpgen_abacus.json` and `examples/water/input_dpgen_cp2k.json`, have been uploaded with the assistance of @zjgemi.

File: examples/water/input_dpgen_abacus.json
Comment (njzjz) [inline examples/water/input_dpgen_abacus.json:1]: > Two example files, `examples/water/input_dpgen_abacus.json` and `examples/water/input_dpgen_cp2k.json`, have been uploaded with the assistance of @zjgemi.

Please add new example files to `tests/check_examples.py`.

**Review by njzjz — CHANGES_REQUESTED** [review]

File: examples/water/input_dpgen_abacus.json
Comment (Andy6M) [reply examples/water/input_dpgen_abacus.json:1]: Thank you for your suggestion. The new example files have been added to `tests/check_examples.py`.

**Review by zjgemi — APPROVED** [review]

**Review by wanghan-iapcm — APPROVED** [review]

**Review by njzjz — APPROVED** [review]

**Pull request merged**